README.md

```diff
@@ -1,2 +1,179 @@
+
+```xml
+                <plugin>
+                    <groupId>org.technologybrewery.baton</groupId>
+                    <artifactId>baton-maven-plugin</artifactId>
+                    <version>0.1.0</version>
+                    <extensions>true</extensions>
+                    <executions>
+                        <execution>
+                            <id>default</id>
+                            <goals>
+                                <goal>baton-migrate</goal>
+                            </goals>
+                        </execution>
+                    </executions>
+                    <dependencies>
+                        <dependency>
+                            <groupId>org.technologybrewery.baton.example</groupId>
+                            <artifactId>example-migration-configuration</artifactId>
+                            <version>1.0.0</version>
+                        </dependency>
+                    </dependencies>
+                </plugin>
+```
+
+### Example result output
+When executing your next Maven build, Baton will execute as part of the `initialize` build lifecycle and output similar
+to the following will result:
+
+```bash
+[INFO] --- baton:0.1.0:baton-migrate (default) @ toml-file-update ---
+[INFO] Loading migrations from: jar:file:/Users/dfh/.m2/repository/org/technologybrewery/baton/example/example-migration-configuration/1.0.0/example-migration-configuration-1.0.0-SNAPSHOT.jar!/migrations.json
+[INFO] Found 1 migrations
+[INFO] Migrations Processed: 1, Successfully Migrated Files: 1, Unsuccessfully Migrated Files: 0
+```
+
+## Configuration
+All Baton configurations may be set either via the `baton-maven-plugin`'s `<configuration>` definition, Maven POM 
+properties, or `-D` on the line and follow a consistent naming pattern for the different configuration approaches.  For 
+setting configurations via POM properties or `-D` on the command line, all configuration keys may be prepended with 
+`baton.`.  For example, `migrationsConfigurationFile` controls the file name that Baton will use to find migrations 
+and may be configured using the following approaches:
+
+1. Plugin `<configuration>`
+
+```xml
+	<plugin>
+		<groupId>org.technologybrewery.baton</groupId>
+		<artifactId>baton-maven-plugin</artifactId>
+		<extensions>true</extensions>
+		<configuration>
+			<migrationsConfigurationFile>alternative-migrations.json</migrationsConfigurationFile>
+		</configuration>
+	</plugin>
+```
+
+2. `-D` via command line
+
+```shell
+mvn clean install -DmigrationsConfigurationFile=alternative-migrations.json
+```
+
+3. POM properties
+
+```xml
+	<properties>
+		<migrationsConfigurationFile>alternative-migrations.json</migrationsConfigurationFile>
+	</properties>
+```
+
+**NOTE:** The above list's order reflects the precedence in which configurations will be applied.  For example, 
+configuration values that are specified in the plugin's `<configuration>` definition will always take precedence, while 
```

Comment:
Could definitely be wrong, but I thought the usual standard was that command line options are higher precedent than POM configuration blocks.  It's kind of nice in things like the release plugin where you can have a reasonable default, but still override it for local testing without making file modifications.

Reply:
Can test this out - was honestly just taking what @ekoniec1 had put together for Habushu.